### PR TITLE
update to use unprefixed Fullscreen API; clean up + DRY (fixes #4311)

### LINKED ISSUE #4311: Call unprefixed Fullscreen API first
[This line](https://github.com/kripken/emscripten/blob/5f091840/src/library_browser.js#L332) causes a deprecation warning in Firefox:

```
Prefixed Fullscreen API is deprecated
```

screen shot 2016-05-11 at 2 52 16 pm

### PR DESCRIPTION
**Pull request by cvan**
need help building it to test

### CODE REVIEW

File: src/library_browser.js [hunk 1]
```diff
@@ -264,6 +264,48 @@ var LibraryBrowser = {
           }, false);
         }
       }
+
+      function Fullscreen (testElement) {
```
Comment (kripken) [inline src/library_browser.js]: convention is to start with lowercase, and no space before `(`.

**Comment by kripken** [comment]
You shouldn't need to build anything to test this, just run the tests with these changes. Something like `./tests/runner.py interactive.test_html5_fullscreen`, and there are a few others with `fullscreen` in their names that are probably relevant too, see `test_interactive.py`.

File: src/library_browser.js (hunk 1 above)
Comment (cvan) [inline src/library_browser.js]: will do

**Comment by juj** [comment]
Also pull requests should go against `incoming` branch. Can you reopen this PR there?

File: src/library_browser.js (hunk 1 above)
Comment (juj) [inline src/library_browser.js]: I'd prefer we don't add new kind of stateful abstractions like this. For avoiding code duplication e.g. on obtaining the current fullscreen element, prefer to have a new function that would read something like

    function fullscreenElement() {
        return document['fullScreenElement'] || document['fullscreenElement'] ||
            document['mozFullScreenElement'] || document['mozFullscreenElement'] ||
            document['msFullScreenElement'] || document['msFullscreenElement'] ||
            document['webkitFullScreenElement'] || document['webkitFullscreenElement'] ||
            document['webkitCurrentFullScreenElement'];
    }

and then call that everywhere we need to get the fullscreen element. That is easier to follow than relate to, than adding wrapper over the fullscreen API.

**Comment by cvan** [comment]
sure thing

**Pull request closed without merging**